### ISSUE TITLE
bug: Dismissing a LoadingController will cause a swiper Slide flicks back to initial slide when swiping and using showing and then dismissing an Ionic Spinner while loading data

### ISSUE THREAD

**Issue by pjc2007** [ionic-team/ionic-framework#29762]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/docs/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v8.x

### Current Behavior

In the sample application when trying to find out why this was happening in my real application.
The basic setup is 2 slides, each with a Angular component, each with some selects, and when swiping from one slide to the next, I show an Ionic spinner while there is a server request for the data of the component in the new slide.

However, I find that if I happen to have opened a select, or any modal popup from within one of the components, and then slide, showing and then hiding the Ionic spinner will cause the slide to go back to the page we just came from.

If I remove the spinner, it does not happen , or we we don't open any "popups" before sliding it does not happen.

The sample app is just a stock newly created Ionic/Angular application, where the main component with the slider is `FolderPage`. In the `slideChanged` handler it calls the `loadData` on the component we are going to, and this is where we call the `presentDataLoading` and `dismissLoaderIfPresent` which uses the Ionic `LoadingControlle`r (which I use everywhere wit no issues)

So, there is some weird interaction between the above elements which is causing the slider to be "pushed" back, but I have no idea what this could be.

### Expected Behavior

To be able to slide to the next slide, and not have it then return to the previous when using an LoadingController during the transition

### Steps to Reproduce

- Run the sample application
- Open a select
- Slide to the next page

Most often when we call the dismiss the slide will then flick back to the orinaly slide

### Code Reproduction URL

https://github.com/pjc2007/swiper-issue2

### Ionic Info

Ionic:

   Ionic CLI                     : 7.2.0 (C:\Users\peter\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 8.2.6
   @angular-devkit/build-angular : 18.1.3
   @angular-devkit/schematics    : 18.1.3
   @angular/cli                  : 18.1.3
   @ionic/angular-toolkit        : 11.0.1

Capacitor:

   Capacitor CLI      : 6.1.1
   @capacitor/android : not installed
   @capacitor/core    : 6.1.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.4
   native-run  : 2.0.1

System:

   NodeJS : v20.15.1 (C:\Program Files\nodejs\node.exe)
   npm    : 10.7.0
   OS     : Windows 10

### Additional Information

I did do an original post about it on SO (https://stackoverflow.com/questions/78833526/swiper-slidechange-event-sometimes-called-a-second-time-resetting-back-to-the-in), but no real feedback there.

Not sure if this is a swiper or Ionic issue,

**Comment by dev-charles15531** [comment]
@pjc2007 Are you able to resolve this yet?

**Comment by pjc2007** [comment]
> @pjc2007 Are you able to resolve this yet?

No, I had to replace it with a custom one, i.e. a component with an `ion-spinner` in it and using an `ngIf` to show/hide. Could not get around the problem not matter what I tried. This causes no issues.

**Comment by thetaPC** [comment]
Verified on 8.8.19 and 9.0.2. Thanks for the reproduction, it made this quick to pin down.

This is not the spinner, it is focus. When you open and close the select, Ionic returns focus to it, which is standard dialog behavior. You then swipe to slide 2 while that select on slide 1 still holds focus. Presenting the loading overlay records the select as the element to restore focus to, and dismissing it focuses the select again. Swiper's accessibility module watches for focus landing in a non-active slide and slides to it, so it takes you back to slide 1. Both behaviors are deliberate, and they cancel each other out here.

Two workarounds, both confirmed against your setup:

Turn off Swiper's scroll-on-focus (needs swiper 11.1.6 or later):

```html
<swiper-container [a11y]="{ scrollOnFocus: false }" events-prefix="swiper-" [modules]="swiperModules" #swiper (swiper-slidechange)="slideChanged($event)">
```

Or blur whatever is focused before presenting, in `presentDataLoading`:

```ts
(document.activeElement as HTMLElement | null)?.blur();
this.loaderBase = await this.loadingService.presentLoadingSpinner('loading');
```

Your current fix, an `ion-spinner` toggled with `ngIf`, works for the same reason: no overlay means no focus restoration.

Closing as a duplicate of #28058, which covers the same interaction.
